frontend/src/pages/HomePage/HomePage.tsx

```diff
@@ -1,12 +1,152 @@
-import GeocacheList from "../../components/GeocacheList";
+import { useState, useEffect } from "react";
+import { getGeoCaches } from "../../api/geocaches";
+import { Container, Card, Row, Button } from "react-bootstrap";
+import "./HomePage.scss";
 
 const HomePage = () => {
+    const [geocaches, setGeocaches] = useState([]);
+
+    useEffect(() => {
+        getGeoCaches().then(geocachesResult => {
+            setGeocaches(geocachesResult.data);
+        });
+    }, []);
+
+    const name1: string = geocaches[0]["name"];
+    const time1: string = geocaches[0]["publishedDate"];
+    const name2: string = geocaches[1]["name"];
+    const time2: string = geocaches[1]["publishedDate"];
+    const name3: string = geocaches[2]["name"];
+    const time3: string = geocaches[2]["publishedDate"];
+
     return (
-        <div>
-            <h1>{"Here is the page's content"}</h1>
-            <GeocacheList/>
-        </div>
+        <>
+            <div className="screen">
+                <div className="main">
+                    <Container>
+                        <Row><div className="mx-auto" style={{ width: "auto" }}>
```

Comment:
This is not a proper way to render a collection of items. What if the amount of geocaches we want to display changes? It shouldn't be hard-coded like this. Instead, a [mapping function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) should be used.

    return (
        <div>
            <h3>Uusimmat geokätköt</h3>
            <Container>
                {
                    geocaches.map((cache) => {
                        return (
                            <Row key={cache.id}>
                                <Card>
                                    <div>{cache.name}</div>
                                    <div>{cache.placedDate}</div>
                                </Card>
                            </Row>
                        )
                    })
                }
            </Container>
        </div >
    );

Reply:
Yes, that looks better. I'll try that

Reply:
Isn't this the same logic of GeocacheList component? Both Home page and GeocacheList are doing the same thing but with a different look. So we could either get rid of GeocacheList or update it and reuse it.